### DISCUSSION TITLE
How to used monailabel(DeepEdit model) to deal with multimodality/multichannel image

### DISCUSSION THREAD

**Discussion by wudalang2023** [Project-MONAI/MONAILabel#1626] [Q&A]
1、	two modalities stacked in a single Nifti file

We have SPECT and CT two modalities stacked in a single Nifti file as follows:

Three two modalities files

SPECT+CT single Nifti file shape (512, 512, 251, 2)
CT+SPECT single Nifti file shape (512, 512, 251, 2)
SPECT+CT single Nifti file shape (2, 512, 512, 251)

Now, We used 3dslicer visualization these files, and compare display effect.

SPECT+CT single Nifti file shape (512, 512, 251, 2)
 
![SPECT+CT single Nifti file shape (512, 512, 251, 2)](https://github.com/Project-MONAI/MONAILabel/assets/129755640/7051a32e-eea8-4af2-94ca-a209db81374b)

CT+SPECT single Nifti file shape (512, 512, 251, 2)

![CT+SPECT single Nifti file shape (512, 512, 251, 2)](https://github.com/Project-MONAI/MONAILabel/assets/129755640/51c2d4a4-38ec-43fe-b8fc-2e7ef8a93de1)

 

SPECT+CT single Nifti file shape (2, 512, 512, 251)

 
![SPECT+CT single Nifti file shape (2, 512, 512, 251)](https://github.com/Project-MONAI/MONAILabel/assets/129755640/a09fb077-d493-4f1a-8bed-99750a561968)

2、	used monailabel(DeepEdit model) to deal with multimodality/multichannel image

https://github.com/Project-MONAI/MONAILabel/discussions/789

Above network link expert think the DeepEdit model can deal with the multimodality. So I think the SPECT+CT single Nifti file shape (512, 512, 251, 2) and CT+SPECT single Nifti file shape (512, 512, 251, 2) files for the DeepEdit model are the same, the result of the segmentation should be the same, but now the result is different. I do not know what is the problem from my method. And I find the result segmentation is the same for SPECT+CT single Nifti file shape (512, 512, 251, 2) and only one modality SPECT image, the CT+SPECT and one modality CT image is the same.

SPECT+CT single Nifti file shape (512, 512, 251, 2)

 
![SPECT+CT single Nifti file shape (512, 512, 251, 2)-seg](https://github.com/Project-MONAI/MONAILabel/assets/129755640/c7ced15c-5ddf-44fa-b497-a5c6a0090702)

CT+SPECT single Nifti file shape (512, 512, 251, 2)

 
![CT+SPECT single Nifti file shape (512, 512, 251, 2)-seg](https://github.com/Project-MONAI/MONAILabel/assets/129755640/4b237e0f-5803-4d57-81bd-9cb11e65aae4)

**Comment by diazandr3s** [comment]
Hi @wudalang2023,

This is a great use case.
I assume both modalitiers are coregistered. You could follow these instructions to make MONAI Label to work on multimodaliy: https://github.com/Project-MONAI/MONAILabel/discussions/1443
